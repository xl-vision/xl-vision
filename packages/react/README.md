# `@xl-vision/react`

> TODO: description

