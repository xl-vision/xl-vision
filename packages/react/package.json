{
  "name": "@xl-vision/react",
  "version": "0.0.0",
  "description": "xl-vision react components.",
  "author": "Rhys Xia <xrs4433@outlook.com>",
  "license": "MIT",
  "main": "legacy/index.js",
  "module": "modern/index.js",
  "types": "legacy/index.d.ts",
  "files": [
    "legacy",
    "modern"
  ],
  "keywords": [
    "xl-vision",
    "react",
    "styles"
  ],
  "homepage": "https://github.com/xl-vision/xl-vision-next/tree/master/packages/react#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/xl-vision/xl-vision-next.git"
  },
  "scripts": {
    "build:legacy": "node ../../scripts/build legacy",
    "build:modern": "node ../../scripts/build modern",
    "build": "npm run build:legacy && npm run build:modern"
  },
  "bugs": {
    "url": "https://github.com/xl-vision/xl-vision-next/issues"
  },
  "dependencies": {
    "@babel/runtime": "^7.12.5",
    "prop-types": "^15.7.2"
  },
  "devDependencies": {
    "@types/react": "^16.8.0",
    "@types/react-dom": "^16.8.0",
    "@xl-vision/styled-engine": "^0.0.0",
    "react": "^16.8.0",
    "react-dom": "^16.8.0"
  },
  "peerDependencies": {
    "@types/react": "^16.8.0 || ^17.0.0",
    "@xl-vision/styled-engine": "^0.0.0",
    "react": "^16.8.0 || ^17.0.0",
    "react-dom": "^16.8.0 || ^17.0.0"
  },
  "peerDependenciesMeta": {
    "@types/react": {
      "optional": true
    }
  },
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  }
}
