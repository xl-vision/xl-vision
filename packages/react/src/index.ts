export { styled, createGlobalStyles } from './styles';
export { default as ThemeProvider } from './ThemeProvider';
export { default as Transition } from './Transition';
export { default as CSSTransition } from './CSSTransition';
export { default as CollapseTransition } from './CollapseTransition';
export { default as TransitionGroup } from './TransitionGroup';
