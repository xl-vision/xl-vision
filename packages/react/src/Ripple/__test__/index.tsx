import { mount } from 'enzyme';
import React from 'react';
import { Ripple, RippleRef } from '@xl-vision/react';

const Demo = ({ leaveAfterEnter }: { leaveAfterEnter?: boolean }) => {
  const rippleRef = React.useRef<RippleRef>(null);

  const events = React.useMemo(() => {
    const start = (e: any) => {
      rippleRef.current?.start(e);
    };
    const stop = () => {
      rippleRef.current?.stop();
    };
    return {
      onMouseDown: start,
      onTouchStart: start,
      onMouseUp: stop,
      onTouchEnd: stop,
      onTouchMove: stop,
      onMouseLeave: stop,
      onDragLeave: stop,
      onBlur: stop,
    };
  }, []);

  return (
    <div className='box' {...events}>
      click me
      <Ripple leaveAfterEnter={leaveAfterEnter} transitionClassName='ripple' ref={rippleRef} />
    </div>
  );
};

describe('Ripple', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  it('test render', () => {
    const wrapper = mount(
      <div>
        click me
        <Ripple transitionClassName='ripple' />
      </div>,
    );
    jest.runAllTimers();

    expect(wrapper.render()).toMatchSnapshot();
  });

  it('test event', () => {
    const wrapper = mount(<Demo />);
    wrapper.find('.box').simulate('mousedown');

    jest.runAllTimers();

    const doms = wrapper.getDOMNode().querySelectorAll<HTMLElement>('.xl-ripple__inner');
    expect(doms.length).toBe(1);
    expect(doms[0].style.display).toBe('');

    wrapper.find('.box').simulate('mouseup');

    jest.runAllTimers();

    expect(wrapper.getDOMNode().querySelectorAll<HTMLElement>('.xl-ripple__inner').length).toBe(0);

    wrapper.find('.box').simulate('blur');
    jest.runAllTimers();

    expect(wrapper.getDOMNode().querySelectorAll<HTMLElement>('.xl-ripple__inner').length).toBe(0);
  });
});
