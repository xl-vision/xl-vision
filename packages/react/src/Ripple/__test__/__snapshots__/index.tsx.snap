// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripple test render 1`] = `
.emotion-0 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 0;
  overflow: hidden;
  border-radius: inherit;
  pointer-events: none;
}

<div>
  <div>
    click me
    <div
      class="xl-ripple emotion-0"
    />
  </div>
</div>
`;
